util/src/main/kotlin/com/kamelia/sprinkler/util/ByteArrayDecoding.kt

```diff
@@ -0,0 +1,150 @@
+@file:JvmName("ByteArrayDecoding")
+
+package com.kamelia.sprinkler.util
+
+import java.nio.ByteOrder
+import java.nio.charset.Charset
+
+/**
+ * Read a [Byte] from the beginning a byte array.
```

Comment:
it's not the beginning anymore if you pass a start, is it ? (same for the other extensions)